src/lib/util/corefile.h

```diff
@@ -61,7 +61,7 @@ class core_file
 	static osd_file::error open_proxy(core_file &file, ptr &proxy);
 
 	// close an open file
-	virtual ~core_file();
+	virtual ~core_file() = default;
```

Comment:
Please don't do this to virtual destructors - it can lead to hard-to-diagnose link errors.  If the virtual destructor has an out-of-line body, the vtable is instantiated at the point where the destructor is defined.  Otherwise the vtable is instantiated when the first virtual member function with an out-of-line body is defined.

The issue is that if you have no out-of-line virtual destructor and you forget to define the first virtual method, you'll get a pile of link errors about the vtable being missing wherever the class is instantiated.  If you have an out-of-line virtual destructor, you'll get a much more helpful error message telling you which member function you forgot to define, at the point where the virtual destructor is defined.

Reply:
There's one such case which I had to revert. I had issues with this in the past as well, but didn't have time to read up on it. I also wanted to file an issue with the clang-tidy people about this as this change causes such issues.